src/pages/index.js

```diff
@@ -24,40 +32,141 @@ const profileEditModalClass = new ModalWithForm(
+    });
+    return data;
+  })
+  .catch((error) => {
+    console.error("Error fetching user info:", error);
+  });
 
-const sectionClass = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
-sectionClass.renderItems();
+let sectionClass;
+
+api
+  .getInitialCards()
+  .then((cardList) => {
+    sectionClass = new Section(
+      { items: cardList, renderer: createCard },
+      ".cards__list"
+    );
+    sectionClass.renderItems();
+  })
+  .catch((error) => {
+    console.error("Error fetching user info:", error);
+  });
 
+// PATCH API
 function handleProfileFormSubmit(formData) {
-  userInfoClass.setUserInfo(formData);
+  profileEditModalClass.setLoading(true, "Save");
+  api
+    .updateProfileInfo(formData.title, formData.description)
+    .then((data) => {
+      userInfoClass.setUserInfo({ title: data.name, description: data.about });
+      return data;
+    })
+    .catch((error) => {
+      console.error("Error fetching user info:", error);
+    })
+    .finally(() => {
+      profileEditModalClass.setLoading(false, "Save");
+    });
 }
 
-function handleImageClick(cardData) {
-  previewImageModalClass.open(cardData);
+function handleProfileAvatar(formData) {
+  profileEditAvatarModalClass.setLoading(true, "Save");
+  api
+    .updateProfileAvatar(formData.avatar)
+    .then((data) => {
+      userInfoClass.setUserAvatar({
+        avatar: data.avatar,
+      });
+      return data;
+    })
+    .catch((error) => {
+      console.error("Error fetching user info:", error);
+    })
+    .finally(() => {
+      profileEditAvatarModalClass.setLoading(false, "Save");
+    });
 }
 
-function createCard(item) {
-  const cardElement = new Card(item, "#card-template", handleImageClick);
-  return cardElement.getCardElement();
+//POST API
+function handleAddCardFormSubmit(obj) {
+  profileAddModalClass.setLoading(true, "Create");
+  const name = obj["title"];
+  const link = obj["Image-link"];
+  api
+    .postCard(name, link)
+    .then((card) => {
+      sectionClass.addItem(card);
+      profileAddForm.reset();
+    })
+    .catch((error) => {
+      console.error("Error fetching user info:", error);
+    })
+    .finally(() => {
+      profileAddModalClass.setLoading(false, "Create");
+    });
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  sectionClass.addItem(cardElement);
+//DELETE API
+function handleDeleteCard(card) {
+  confirmationModalClass.open();
+  confirmationModalClass.setSubmitAction(() => {
+    api
+      .deleteCard(card._id)
+      .then((data) => {
+        card.handleDeleteButton();
+      })
+      .catch((error) => {
+        console.error("Error fetching user info:", error);
+      });
+  });
 }
 
-function handleAddCardFormSubmit(obj) {
-  const name = obj["title"];
-  const link = obj["Image-link"];
-  renderCard({ name, link });
-  profileAddForm.reset();
+//PUT API / DELETE
+function handleLikeCard(card) {
+  if (card._isLiked) {

```

Comment:
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.